orderer/consensus/hlmirbft/chain.go

```diff
@@ -591,6 +592,102 @@ func (c *Chain) writeBlock(block *common.Block, index uint64) {
+	if err != nil {
+		return nil,err
+	}
+	// get config update
+	configUpdate, err := configtx.UnmarshalConfigUpdateFromPayload(payload)
+	if err != nil {
+		return nil,err
+	}
+	return  MetadataFromConfigUpdate(configUpdate)
+
+}
+//JIRA FLY2-103 : function to update protocol parameters
+func (c *Chain) updateProtocolParameters(configMetaData *hlmirbft.ConfigMetadata,nodeList []uint64)  {
+	c.Node.config.HeartbeatTicks = configMetaData.Options.HeartbeatTicks
+	c.Node.config.SuspectTicks = configMetaData.Options.SuspectTicks
+	c.Node.config.NewEpochTimeoutTicks = configMetaData.Options.NewEpochTimeoutTicks
+	c.Node.config.BufferSize = configMetaData.Options.BufferSize
+
+	c.opts.HeartbeatTicks = configMetaData.Options.HeartbeatTicks
+	c.opts.SuspectTicks = configMetaData.Options.SuspectTicks
+	c.opts.NewEpochTimeoutTicks = configMetaData.Options.NewEpochTimeoutTicks
+	c.opts.BufferSize = configMetaData.Options.BufferSize
+	c.mirbftMetadataLock.Lock()
+	c.opts.BlockMetadata.ConsenterIds = nodeList
+	c.mirbftMetadataLock.Unlock()
+
+}
+//JIRA FLY2-103 : function to generate new network state config
+func (c *Chain) getNewNetworkStateConfig(newNodeList []uint64) *msgs.NetworkState_Config  {
+	nodes := newNodeList
+	nodeCount := len(nodes)
+	numberOfBuckets := int32(nodeCount)
+	checkpointInterval := numberOfBuckets * 5
+	maxEpochLength := uint64(checkpointInterval * 10)
+	return &msgs.NetworkState_Config{
+			Nodes:              nodes,
+			MaxEpochLength:     maxEpochLength,
+			CheckpointInterval: checkpointInterval,
+			F:                  int32((nodeCount - 1) / 3),
+			NumberOfBuckets:    numberOfBuckets,
+		}
+}
+
+//JIRA FLY2-103 : function to identify the type of config update and return the config change
+func (c *Chain) getUpdatedConfigChange(configMetaData *hlmirbft.ConfigMetadata,currentConsenters , updatedConsenters []*hlmirbft.Consenter) ([]*msgs.Reconfiguration,error){
+	var updatedConfig,newNetworkState *msgs.Reconfiguration
+	var newNetStateConfig *msgs.NetworkState_Config
+	configChangeType := len(currentConsenters) - len(updatedConsenters)
+	consenterList := c.opts.BlockMetadata.ConsenterIds
+	if configChangeType > 0 {
+		newNodeId := uint64(len(currentConsenters)+1)
+		updatedConfig.Type = &msgs.Reconfiguration_NewClient_{NewClient: &msgs.Reconfiguration_NewClient{
+			Id: newNodeId,
+			Width: 100,
+		}}
+		consenterList = append(consenterList,newNodeId)
+		newNetStateConfig = c.getNewNetworkStateConfig(consenterList)
+	} else if configChangeType < 0 {
+		removedConsenter := CompareConsenterList(currentConsenters,updatedConsenters)
+		removedConsenterID,ok := GetConsenterId(c.opts.Consenters,removedConsenter)
+		if !ok {
+			return nil, errors.Errorf("Cannot Retrive Consenter ID")
+		}
+		updatedConfig.Type = &msgs.Reconfiguration_RemoveClient{
+			RemoveClient: removedConsenterID,
+		}
+		consenterList = removeNodeID(consenterList,removedConsenterID)
+		newNetStateConfig = c.getNewNetworkStateConfig(consenterList)
```

Comment:
you can replace this call and the other in the if statement with one outside of the conditional flow.